modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/message/body/MultipartBodyMessageBodyReader.java

```diff
@@ -66,11 +70,35 @@ public MultipartBody readFrom(
 		MediaType mediaType, MultivaluedMap<String, String> multivaluedMap,
 		InputStream inputStream) {
 
+		Message message = JAXRSUtils.getCurrentMessage();
+
+		UploadServletRequest uploadServletRequest = _getUploadServletRequest(
```

Comment:
Would it be possible to use [_portal.getUploadServletRequest](https://github.com/liferay/liferay-portal/blob/04aa57bf2c79629b8f3bff76e9102e882a8e01f7/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L5063) instead?

Reply:
No, it is not possible. This is [the reason](https://github.com/liferay/liferay-portal/blob/04aa57bf2c79629b8f3bff76e9102e882a8e01f7/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L5081).

As the ServletRequest is wrapped multiple times, the wrappers are:

com.liferay.portal.servlet.AuthVerifierServletRequest
org.eclipse.equinox.http.servlet.internal.servlet.HttpServletRequestWrapperImpl
[...]

As the second class name does not start with "com.liferay.", it is considered it must not be unwrapped, and then, the UploadServletRequest is built again, so no multiparts are found as the InputStream is already closed

Reply:
This probably happens because other "external" filters are executed in the middle, so some wrappers are wrapped by internal and external wrappers

Reply:
But good catch :+1:

Reply:
ok, makes sense.